oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/query/async/ElasticResultRowAsyncIterator.java

```diff
@@ -272,7 +274,9 @@ public void onSuccess(SearchResponse<ObjectNode> searchResponse) {
                 LOG.debug("Processing search response that took {} to read {}/{} docs", searchResponse.took(), hitsSize, totalHits);
                 lastHitSortValues = searchHits.get(hitsSize - 1).sort();
                 scannedRows += hitsSize;
-                anyDataLeft.set(totalHits > scannedRows);
+                if (searchResponse.hits().total().relation() == TotalHitsRelation.Eq) {
+                    anyDataLeft.set(totalHits > scannedRows);
+                } else anyDataLeft.set(true);
```

Comment:
I find the following more easy to read: 

    } else {
        anyDataLeft.set(true);
    }

Reply:
done